File: gobrui1/FSharp/src/GameObject/Token/Prefab/Effect/Explosion.fs
namespace ExplosionF

open UnityEngine
open TokenF
open tokenUtil
open PrefabCount
open TeamF
open System.Runtime
open System.Threading.Tasks
open System.Threading
open System.Collections

/// パーティクル
type Explosion() =
    inherit Token()

    member this.StartFunc = this.ShrinkOut 0.8f

    /// コルーチンでフェードアウト(CSharpで設定してる)
    static member Add(attacktarget: Character) =
        let prefab = GetPrefab null PrefabCount.Explosion
        let expro = CreateInstance2<Explosion>(prefab, attacktarget.tokenX, attacktarget.tokenY, "Explosion")
        expro

File: gobrui1/Assets/Scripts/GameObject/Prefab/Effect/Explosion.cs
using Microsoft.FSharp.Control;
using Microsoft.FSharp.Core;
using System.Threading.Tasks;
using System.Threading;
using UnityEngine;
using UniRx.Async;
using System.Collections;
/// パーティクル
public class Explosion : ExplosionF.Explosion
{
    public IEnumerator Start()
    {
        return this.StartFunc;
    }
}